src/main/java/org/stellar/sdk/TransactionBuilder.java

```diff
@@ -0,0 +1,231 @@
+        if (mMemo != null) {
+            throw new RuntimeException("Memo has been already added.");
+        }
+        checkNotNull(memo, "memo cannot be null");
+        mMemo = memo;
+        return this;
+    }
+
+    /**
+     * Adds a <a href="https://developers.stellar.org/docs/glossary/transactions/" target="_blank">time-bounds</a> to this transaction.
+     *
+     * @param timeBounds tx can be accepted within this time bound range
+     * @return Builder object so you can chain methods.
+     * @see TimeBounds
+     * @deprecated this method will be removed in upcoming releases, use <code>addPreconditions()</code> instead for more control over preconditions.
+     */
+    public TransactionBuilder addTimeBounds(TimeBounds timeBounds) {
+        checkNotNull(timeBounds, "timeBounds cannot be null");
+        if (mPreconditions.getTimeBounds() != null) {
+            throw new RuntimeException("TimeBounds already set.");
+        }
+
+        mPreconditions = mPreconditions.toBuilder().timeBounds(timeBounds).build();
+        return this;
+    }
+
+    /**
+     * Because of the distributed nature of the Stellar network it is possible that the status of your transaction
+     * will be determined after a long time if the network is highly congested.
+     * If you want to be sure to receive the status of the transaction within a given period you should set the
+     * {@link TimeBounds} with <code>maxTime</code> on the transaction (this is what <code>setTimeout</code> does
+     * internally; if there's <code>minTime</code> set but no <code>maxTime</code> it will be added).
+     * Call to <code>Builder.setTimeout</code> is required if Transaction does not have <code>max_time</code> set.
+     * If you don't want to set timeout, use <code>TIMEOUT_INFINITE</code>. In general you should set
+     * <code>TIMEOUT_INFINITE</code> only in smart contracts.
+     * Please note that Horizon may still return <code>504 Gateway Timeout</code> error, even for short timeouts.
+     * In such case you need to resubmit the same transaction again without making any changes to receive a status.
+     * This method is using the machine system time (UTC), make sure it is set correctly.
+     *
+     * @param timeout Timeout in seconds.
+     * @return updated Builder
+     * @see TimeBounds
+     * @deprecated this method will be removed in upcoming releases, use <code>addPreconditions()</code> with TimeBound
+     * set instead for more control over preconditions.
+     */
+    public TransactionBuilder setTimeout(long timeout) {
+        if (mPreconditions.getTimeBounds() != null && mPreconditions.getTimeBounds().getMaxTime() != TransactionPreconditions.TIMEOUT_INFINITE) {
+            throw new RuntimeException("TimeBounds.max_time has been already set - setting timeout would overwrite it.");
+        }
+
+        if (timeout < 0) {
+            throw new RuntimeException("timeout cannot be negative");
+        }
+
+        mTimeoutSet = true;
```

Comment:
Is this variable used elsewhere? Why isn't it checked on line 153 above?

Reply:
thanks for drilling into that, reviewed this area and cleaned up considerably, was able to remove this usage.